synapse/handlers/stats.py

```diff
@@ -84,6 +84,10 @@ def _unsafe_process(self):
         # Loop round handling deltas until we're up to date
 
         while True:
+            room_max_stream_ordering = yield self.store.get_room_max_stream_ordering()
```

Comment:
```suggestion
            # Be sure to read the max stream_ordering *before* checking if there are any outstanding
            # deltas, since there is otherwise a chance that we could miss updates which arrive
            # after we check the deltas.
            room_max_stream_ordering = yield self.store.get_room_max_stream_ordering()
```